src/listener.py

```diff
@@ -0,0 +1,190 @@
+import select
+import socket
+from pascy.l2 import *
+import sys
+
+MY_ADDRESSES = {
+	"net1": ["02:42:37:94:2b:e4", "1.1.1.1"],
+	"net2": ["02:42:04:89:4c:a0", "2.2.2.1"]
+	}
+
+ROUTING_TABLE = {
+	"net1": 
+			{ "1.1.1.2": "02:42:01:01:01:02" 
+			},
+	"net2":
+			{ "2.2.2.2": "02:42:02:02:02:02"
+			}
+	}
+
+PACKETS_TO_SEND = []
+
+def get_network_by_ip(ip):
+	for network in ROUTING_TABLE:
+		if ip in ROUTING_TABLE[network].keys():
+			return network
+	return ""
+
+def verify_addresses(ip_address, mac_address):
+	
+	if ip_address in ROUTING_TABLE["net1"].keys():
+			# For regular Router behavior
+			if ROUTING_TABLE["net1"][ip_address] != mac_address:
+				ROUTING_TABLE["net1"][ip_address] = mac_address
+			return True
+		
+	elif ip_address in ROUTING_TABLE["net2"].keys():
+		# For regular Router behavior
+		if ROUTING_TABLE["net2"][ip_address] != mac_address:
+			ROUTING_TABLE["net2"][ip_address] = mac_address
+		return True
+	else:
+		return False
+
+def parse_buffer(buffer):
+	# Ethernet part
+	ethernet_layer = EthernetLayer()
+	ethernet_layer.deserialize(buffer[:14])
+
+	destination_mac = MacAddress.mac2str(ethernet_layer.fields["dst"].get())
+	ethernet_type = ethernet_layer.fields["ether_type"].get()
+
+	if destination_mac == MAC_BROADCAST and ethernet_type == 0x806:
+		parse_ARP(buffer[14:], ethernet_layer)
+
+	elif ethernet_type == 0x800:
+		parse_IP(buffer, ethernet_layer)
+
+
+def parse_IP(buffer, ethernet_layer):
+	#IP part
+	ip_layer = IPLayer()
+	ip_layer.deserialize(buffer[14:34])
+
+	# Calculate the checksum and verify it.
+	if IPLayer.checksum(buffer[14:34]) == ip_layer.fields["checksum"].get():
+
+		destination_ip = IPAddress.ip2str(ip_layer.fields["dst"].get())
+		source_ip = IPAddress.ip2str(ip_layer.fields["src"].get())
+
+		if not verify_addresses(source_ip, MacAddress.mac2str(ethernet_layer.fields["src"].get())):
+			return
+
+		if destination_ip == MY_ADDRESSES["net1"][1] or destination_ip == MY_ADDRESSES["net2"][1]:
+			pass
+			# The destination is ME!
+			# if ip_layer.fields["protocol"].get()
+
+		# If the packet need to be forward
+		elif destination_ip in ROUTING_TABLE["net1"].keys() or destination_ip in ROUTING_TABLE["net2"].keys():
```

Comment:
Consider
```python
routes = []
routes.extend(ROUTING_TABLE["net1"].keys())
routes.extend(ROUTING_TABLE["net2"].keys())

elif destination_ip in routes:
```
It makes it more readable, especially if there's more than two items